### PR TITLE
test: added interaction mock with review transaction tests

### LINKED ISSUE #104: Add interactions mock to use in Cypress
We need to mock interactions that comes from dapp to wallet. Possibly we could reuse WebSocket mock approach that is used in `frontend-monorepo`. 

Event logic with example added - https://github.com/vegaprotocol/vegawallet-ui/pull/110
Need to provide Cypress functions that use it properly with flows descriptions and type definitions.
Cover https://github.com/vegaprotocol/specs/blob/master/user-interface/0001-WALL-wallet.md#0001-WALL-024 and https://github.com/vegaprotocol/specs/blob/master/user-interface/0001-WALL-wallet.md#0001-WALL-025 ACs in this chore.

### PR DESCRIPTION
**Pull request by radoslaw-szpiech**
# Related issues 🔗

Closes #104 

# Description ℹ️

Added Cypress function that trigger Transaction Review event.
Covered [0001-WALL-024](https://github.com/vegaprotocol/specs/blob/master/user-interface/0001-WALL-wallet.md#0001-WALL-024) and [0001-WALL-025](https://github.com/vegaprotocol/specs/blob/master/user-interface/0001-WALL-wallet.md#0001-WALL-025).

### CODE REVIEW

File: apps/wallet-mock/src/support/event-trigger.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    win.document.body.dispatchEvent(
+      new CustomEvent('new_interaction', {
+        detail: {
+          name: 'INTERACTION_SESSION_BEGAN',
+          data: { workflow: 'TRANSACTION_REVIEW' },
+          traceID: '1',
+        },
+      })
+    )
+  })
```
Comment (dexturr) [inline apps/wallet-mock/src/support/event-trigger.ts:14]: Only needs to be done before a single workflow not every single time

File: apps/wallet-mock/src/support/event-trigger.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  cy.window().then((win) => {
+    win.document.body.dispatchEvent(
+      new CustomEvent('new_interaction', {
+        detail: {
+          name: 'INTERACTION_SESSION_ENDED',
+          traceID: '1',
+        },
+      })
+    )
+  })
```
Comment (dexturr) [inline apps/wallet-mock/src/support/event-trigger.ts:35]:
```
cy.sendBackendInteraction = (type, data) => {
  cy.window().then((win) => {
    win.document.body.dispatchEvent(
      new CustomEvent('new_interaction', {
        detail: data
      })
    )
  })
}
```

File: apps/wallet-mock/src/support/event-trigger.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
+// For now it supports only simple transaction review event, ideally it should be extended to support all possible workflows
+
+Cypress.Commands.add('triggerTransactionEvent', (event, data) => {
+  cy.window().then((win) => {
+    win.document.body.dispatchEvent(
+      new CustomEvent('new_interaction', {
+        detail: {
+          name: 'INTERACTION_SESSION_BEGAN',
+          data: { workflow: 'TRANSACTION_REVIEW' },
+          traceID: '1',
```
Comment (dexturr) [inline apps/wallet-mock/src/support/event-trigger.ts:10]: Trace id's should be unique to mirror backend, can we use nanoid to generate these

File: apps/wallet-mock/src/support/event-trigger.ts (hunk 1 above)
Comment (radoslaw-szpiech) [reply apps/wallet-mock/src/support/event-trigger.ts:14]: Make sense. I will divide into 3 separate cypress functions

**Review by dexturr — APPROVED** [review]

**Pull request merged**